src/main.cc

```diff
@@ -67,12 +68,58 @@ void process_event(const std::shared_ptr<event::BaseEvent> &event) {
               << (install_complete->success ? "success" : "failure") << "\n";
   } else if (event->variant == "UpdateCheckComplete") {
     const auto check_complete = dynamic_cast<event::UpdateCheckComplete *>(event.get());
-    std::cout << check_complete->result.updates.size() << " updates available\n";
+    switch (check_complete->result.status) {
+    case result::UpdateStatus::kError:
+      std::cout << check_complete->result.updates.size() << " updates with error";
```

Comment:
Similarly, if there is an error, I think no updates are returned, so there's no need to print a number, and possibly no need to print anything at all.